src/com/lftechnology/java/training/alina/jdbc/dao/user/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+            ResultSet result = preparedStatement.executeQuery();
+            while (result.next()) {
+                user.setUsername(result.getString("username"));
+                user.setTerminated(result.getBoolean("is_terminated"));
+                user.setCreatedAt(result.getTimestamp("created_at"));
+                userList.add(user);
+            }
+            preparedStatement.close();
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { e });
+        }
+        return userList;
+    }
+
+    @Override
+    public User addNew(User user) {
+        String sql = "Insert into user (username,password,is_terminated,created_at) values (?,?,?,?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, user.getUsername());
+            preparedStatement.setString(2, user.getPassword());
+            preparedStatement.setBoolean(3, Constants.NOT_TERMINATED);
+            preparedStatement.setTimestamp(4, user.getCreatedAt());
+            int result = preparedStatement.executeUpdate();
+            ResultSet rs = preparedStatement.getGeneratedKeys();
+            if (result > 0) {
+                if (rs.next()) {
+                    user.setUserId(rs.getInt(1));
+                }
+            }
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return user;
+    }
+
+    @Override
+    public Boolean delete(String fullname) {
+        boolean isDeleted = false;
+        String sql =
+                "update user u inner join employee e on u.user_id=e.user_id set u.is_terminated=?,u.modified_at=?,e.modified_at=? where (e.fullname=? and u.is_terminated=?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setBoolean(1, Constants.TERMINATED);
+            preparedStatement.setTimestamp(2, DateTimeService.getCurrentTimeStamp());
+            preparedStatement.setTimestamp(3, DateTimeService.getCurrentTimeStamp());
+            preparedStatement.setString(4, fullname);
+            preparedStatement.setBoolean(5, Constants.NOT_TERMINATED);
+            int result = preparedStatement.executeUpdate();
+            isDeleted = (result != 0) ? true : false;
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return isDeleted;
+    }
+
+    @Override
+    public List<User> searchUser(String sql, String... searchContent) {
+        return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)